File: backend/src/main/kotlin/com/cyborgstudios/gamebuilder/orchestration/domain/repository/ArtifactRepository.kt
package com.cyborgstudios.gamebuilder.orchestration.domain.repository

import com.cyborgstudios.gamebuilder.orchestration.domain.model.Artifact
import com.cyborgstudios.gamebuilder.orchestration.domain.model.ArtifactType
import java.util.UUID

interface ArtifactRepository {

    fun save(artifact: Artifact): Artifact

    fun findById(id: UUID): Artifact?

    fun findByProjectId(projectId: UUID): List<Artifact>

    fun findByProjectIdAndType(projectId: UUID, type: ArtifactType): Artifact?
}

File: backend/src/gamebuilder/orchestration/domain/repository/artifact_repository.py
from typing import Protocol
from uuid import UUID

from gamebuilder.orchestration.domain.model.artifact import Artifact


class ArtifactRepository(Protocol):
    async def save(self, artifact: Artifact) -> Artifact: ...

    async def find_by_project_id(self, project_id: UUID) -> list[Artifact]: ...
